addons/pos_self_order/static/src/kiosk/self_order_kiosk_service.js

```diff
@@ -12,13 +13,16 @@ export class selfOrder extends selfOrderCommon {
     }
 
     async setup(...args) {
-
         await super.setup(...args);
 
         this.bus_service.addChannel(`pos_config-${this.access_token}`);
         this.bus_service.addEventListener("notification", ({ detail }) => {
             for (const message of detail) {
-                this.dispatchMessage(message);
+                if (message.type === "PRODUCT_CHANGED") {
+                    this.handleProductChanges(message.payload);
+                } else {
+                    this.dispatchMessage(message);
+                }
```

Comment:
Normally, you would modify the `displayMessage` function rather than here to filter on message type.